Source/OxyPlot.SkiaSharp.Avalonia/DoubleBuffered/PlotRenderer.cs

```diff
@@ -0,0 +1,225 @@
+            {
+                this.backBuffer?.Dispose();
+                this.backBuffer = new SKBitmap(width, height, SKColorType.Rgba8888, SKAlphaType.Premul);
+            }
+
+            return dpiScale;
+        }
+
+        /// <summary>
+        /// This loop runs until canceled and updates and renders the plot if required.
+        /// </summary>
+        private void RenderLoop(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                this.renderRequiredEvent.WaitOne();
+                cancellationToken.ThrowIfCancellationRequested();
+                var size = this.Bounds.Size;
+
+                if (size.Width > 0 && size.Height > 0 && this.PlotView.ActualModel is PlotModel plotModel)
+                {
+                    var isUpdateRequired = Interlocked.Exchange(ref this.PlotView.isUpdateRequired, 0);
+                    var iPlotModel = (IPlotModel)plotModel;
+                    if (isUpdateRequired > 0)
+                    {
+                        lock (plotModel.SyncRoot)
+                        {
+                            iPlotModel.Update(isUpdateRequired > 1);
+                            cancellationToken.ThrowIfCancellationRequested();
+                            this.Render(iPlotModel, size);
+                        }
+                    }
+                    else if (Interlocked.Exchange(ref this.PlotView.isRenderRequired, 0) == 1)
+                    {
+                        lock (plotModel.SyncRoot)
+                        {
+                            this.Render(iPlotModel, size);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void Render(IPlotModel model, Size size)
+        {
+            var scale = this.EnsureBackBuffer(size);
+
+            using (var canvas = new SKCanvas(this.backBuffer))
+            {
+                canvas.Scale((float)scale);
+                this.renderContext.SkCanvas = canvas;
+                canvas.Clear(model.Background.ToSKColor());
+                model.Render(this.renderContext, new OxyRect(0, 0, this.backBuffer.Width / scale, this.backBuffer.Height / scale));
+            }
+
+            this.BufferSwitch();
+            Dispatcher.UIThread.InvokeAsync(this.InvalidateVisual);
+        }
+
+        /// <inheritdoc />
+        protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToVisualTree(e);
+            this.StartRenderLoop();
+        }
+
+        /// <inheritdoc />
+        protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnDetachedFromVisualTree(e);
+            this.StopRenderLoop();
+        }
+
+        private void StartRenderLoop()
+        {
+            new Task(() =>
```

Comment:
Because all this stuff is deferred, I think it's possible for it to be started by being attached, and then be detached before it gets around to creating the CTS, and so run when not attached.